### PR TITLE
Doc: update documentation for on_bytes

### PR DESCRIPTION
**Pull request by chentoast**
Plus generated some extra documentation that might have been from earlier PRs.

@bfredl

### CODE REVIEW

File: src/nvim/api/buffer.c [hunk 1]
```diff
@@ -111,6 +111,24 @@ Integer nvim_buf_line_count(Buffer buffer, Error *err)
...
+///               - the string "bytes"
+///               - buffer handle
+///               - b:changedtick
+///               - start row of the changed text (zero-indexed)
+///               - start column of the changed text
+///               - byte offset of the changed text (from the start of
+///                   the buffer)
+///               - old end row of the changed text
+///               - old end column of the changed text
+///               - old end byte offset of the changed text
```
Comment (bfredl) [inline src/nvim/api/buffer.c:128]: byte length

File: src/nvim/api/buffer.c [hunk 2]
```diff
@@ -111,6 +111,24 @@ Integer nvim_buf_line_count(Buffer buffer, Error *err)
...
+///               - new end row of the changed text
+///               - new end column of the changed text
+///               - new end byte offset of the changed text
```
Comment (bfredl) [inline src/nvim/api/buffer.c:131]: ditto

**Comment by bfredl** [comment]
otherwise LGTM :+1:

**Comment by chentau** [comment]
@bfredl Done :)

**Comment by bfredl** [comment]
thanks, merged :)

**Pull request merged**